File: app/src/main/java/com/anokmik/tripassistant/user/profile/ProfileFragment.java
package com.anokmik.tripassistant.user.profile;

import android.os.Bundle;

import com.anokmik.tripassistant.R;
import com.anokmik.tripassistant.base.BaseFragment;
import com.anokmik.tripassistant.databinding.FragmentProfileBinding;
import com.anokmik.tripassistant.trip.Key;

public final class ProfileFragment extends BaseFragment<FragmentProfileBinding> implements ProfileContract.View {

    public static ProfileFragment newInstance(long userId) {
        Bundle args = new Bundle();
        args.putLong(Key.USER_ID, userId);
        ProfileFragment fragment = new ProfileFragment();
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    protected boolean displayHomeAsUp() {
        return true;
    }

    @Override
    protected int getTitleResourceId() {
        return R.string.title_profile;
    }

    @Override
    protected int getLayoutId() {
        return R.layout.fragment_profile;
    }

    @Override
    protected void initBinding(FragmentProfileBinding binding) {
        binding.setPresenter(new ProfilePresenter(this, getUserId()));
    }

    @Override
    public void back() {
        getActivity().onBackPressed();
    }

    private long getUserId() {
        return getArguments().getLong(Key.USER_ID);
    }

}

File: app/src/main/kotlin/com/anokmik/tripassistant/user/profile/ProfileFragment.kt
package com.anokmik.tripassistant.user.profile

import android.os.Bundle

import com.anokmik.tripassistant.R
import com.anokmik.tripassistant.base.BaseFragment
import com.anokmik.tripassistant.databinding.FragmentProfileBinding
import com.anokmik.tripassistant.trip.USER_ID

class ProfileFragment : BaseFragment<FragmentProfileBinding>(), ProfileContract.View {

    override val displayHomeAsUp = true

    override val layoutId = R.layout.fragment_profile

    override val titleResourceId = R.string.title_profile

    private val userId by lazy { arguments.getLong(USER_ID) }

    override fun initBinding(binding: FragmentProfileBinding) {
        binding.presenter = ProfilePresenter(this, userId)
    }

    override fun back() = activity.onBackPressed()

    companion object {

        fun newInstance(userId: Long): ProfileFragment {
            val args = Bundle()
            args.putLong(USER_ID, userId)
            val fragment = ProfileFragment()
            fragment.arguments = args
            return fragment
        }

    }

}